Code/Framework/AzToolsFramework/AzToolsFramework/UI/Outliner/EntityOutlinerWidget.cpp

```diff
@@ -1206,6 +1206,12 @@ namespace AzToolsFramework
         });
     }
 
+    void EntityOutlinerWidget::OnPrefabTemplateDirtyFlagUpdated(
+        [[maybe_unused]] Prefab::TemplateId templateId, [[maybe_unused]] bool status)
+    {
+        m_gui->m_objectTree->update();
```

Comment:
Honestly, I was thinking about this too but I was conflicted whether trying to optimize this was actually worth it.

There could be multiple instances of each prefab template in the model. So the notification should
- Retrieve all instances for the template
- Get the container entity for each instance
- Convert that entityId to a QModelIndex
- trigger dataChanged for each index

I'm not sure if this is actually faster than just updating the view, especially as I would expect update() to only refresh the visible portion of the view? (Or if it doesn't, then maybe that could be the optimization).

Reply:
Does this mean the entire outliner for a large level will get refreshed when a single leaf prefab is saved? If that's the case, we should think of a better way to do this because making any changes on large levels is already slow and this will be adding more to that slowness.

Just to get an eyeball of how long this tree update will contribute, can you try to update a prefab in AutomatedTesting\Levels\Performance\10KEntityCpuPerfTest and profile how much time this method takes? If it's anything more than a second, we should reevaluate this approach

Reply:
As I said before, I would expect update() to only refresh the visible portion of the view. I will try to verify if that's the case.